src/blog/2024/04/displaying-logged-in-users-on-dashboard.md

```diff
@@ -0,0 +1,142 @@
+   "username": "", // FlowFuse Username
+   "email": "", // E-Mail Address connected to their FlowFuse account
+   "name": "", // Full Name
+   "image": "" // User Avatar from FlowFuse
+}
+```
+
+Behind the scenes, the user addon is appending the user object to the `msg`, via the SocketIO auth option. We make the socketio object available via a computed [setup](https://dashboard.flowfuse.com/contributing/guides/state-management.html#setup-store) object, this means that we can also access user data in any ui-template widget like this:
+
+```
+setup.socketio.auth.user
+```
+When running Node-RED Dashboard 2.0 on FlowFuse, you'll have a new sidebar option in the Node-RED Editor, you just have to navigate to the FF Auth tab and you’ll see two option
+
+**Option 1: Include client Data**
+
+By default, this option is enabled. When this option is enabled, an object with user information will be added to the “msg” emitted by any widget of the Node-red Dashboard 2.0.
+
+**Option 2: Accept client constraints**
+
+A feature that ensures messages are specifically targeted to individual clients, which enhances the precision and security of data transmission within the platform. It determines by enabling the nodes option in the FF Auth tab whether the enabled node type will utilize client data, such as socketid, and restrict communications to only that client.
+
+For example, consider a manufacturing facility where each production line has its own monitoring system. With this feature enabled, data from sensors on Production Line A will only be sent to the monitoring system designated for Production Line A. This ensures that data remains isolated and relevant to each specific area of operation, maintaining organizational efficiency and security.
+
+*Note: Please note that Multi-User Addons can only be used by our Teams and Enterprise Self-Hosted customers. Upon request, we provide all required configurations to get started.*
+
+!["Screenshot displaying the Flowfuse Muti-user addon option"](./images/displaying-logged-in-user-ff-auth-tab.png "Screenshot displaying the Flowfuse Muti-user addon option")
+
+## Displaying logged in user on Dashboard 2.0
+
+Now you know how the user add on works, you are all set to display logged in user on Dashboard 2.0. To confirm this you can use a `debug` node that receives the `msg` object emitted by the Dashboard 2.0 widgets.
+
+To display user information on the dashboard we will use Vue’s [Teleport](https://dashboard.flowfuse.com/nodes/widgets/ui-template.html#teleports) feature to render content to a specific location in the DOM, we will display user information at the action bar’s right-hand side.
+
+1. Drag an ui-template widget onto the canvas.
+2. Click on that node, and select type as “Widget (Ui-Scoped)”. ( this allows us to render this ui-template at ui scoped which means I will not required to add separate ui-templates for different pages )
+3. Copy the below vue snippet and paste that into the ui-template.
+
+```
```

Comment:
```suggestion
```vue
```

Reply:
@joepavitt Currently vue is not supported by Prism, here is the list of supported languages https://prismjs.com/#languages-list